internal/core/src/query/RangeSearchHelper.h

```diff
@@ -0,0 +1,133 @@
+     *      { i(0,0), i(0,1), …, i(0,k0-1),
+     *        i(1,0), i(1,1), …, i(1,k1-1),
+     *          …,
+     *        i(n-1,0), i(n-1,1), …, i(n-1,kn-1)},
+     *      i(0,0), i(0,1), …, i(0,k0-1) means the ids of RangeSearch result querys[0], k0 equals lim[1] - lim[0];
+     * dist: the size of ids is lim[nq],
+     *      { d(0,0), d(0,1), …, d(0,k0-1),
+     *        d(1,0), d(1,1), …, d(1,k1-1),
+     *          …,
+     *        d(n-1,0), d(n-1,1), …, d(n-1,kn-1)},
+     *      d(0,0), d(0,1), …, d(0,k0-1) means the distances of RangeSearch result querys[0], k0 equals lim[1] - lim[0];
+     */
+    auto lims = knowhere::GetDatasetLims(data_set);
+    auto id = knowhere::GetDatasetIDs(data_set);
+    auto dist = knowhere::GetDatasetDistance(data_set);
+    std::cout << "lxg debug after" << std::endl;
+
+    // use p_id and p_dist to GenResultDataset after sorted
+    auto p_id = new int64_t[topk * nq];
+    auto p_dist = new float[topk * nq];
+    //  cnt means the subscript of p_id and p_dist
+    int cnt = 0;
+
+    for (int i = 0; i < nq; i++) {
+        // if RangeSearch answer size of one nq is less than topk, set the capacity to size
+        auto size = lims[i + 1] - lims[i];
+        int capacity = topk > size ? size : topk;
+        /*
+         *   get result for one nq
+         *   IP:   1.0        range_filter     radius
+         *          |------------+---------------|       min_heap   descending_order
+         *   L2:   0.0        range_filter     radius
+         *          |------------+---------------|       max_heap   ascending_order
+         *
+         */
+        if (IsMetricType(metric_type, knowhere::metric::IP)) {
+            MinHeap sub_result;
+            for (int j = lims[i]; j < lims[i + 1]; j++) {
+                auto current = std::pair(dist[j], id[j]);
+                if (sub_result.size() == capacity) {
+                    if (sub_result.top().first > current.first) {
+                        current = sub_result.top();
+                    }
+                    sub_result.pop();
+                }
+                sub_result.push(current);
+            }
+            for (int i = cnt; i < capacity + cnt; i++) {
+                p_dist[i] = sub_result.top().first;
+                p_id[i] = sub_result.top().second;
+            }
+
+        } else {
+            MaxHeap sub_result;
+
+            for (int j = lims[i]; j < lims[i + 1]; j++) {
+                auto current = std::pair(dist[j], id[j]);
+                if (sub_result.size() == capacity) {
+                    if (sub_result.top().first < current.first) {
+                        current = sub_result.top();
+                    }
+                    sub_result.pop();
+                }
+                sub_result.push(current);
+            }
+            for (int i = cnt; i < capacity + cnt; i++) {
+                p_dist[i] = sub_result.top().first;
+                p_id[i] = sub_result.top().second;
```

Comment:
1. This is a max-heap, so need to iterate reversely from `capacity+cnt-1` to `cnt`
2. After the assignment, need to pop the top result